lib/Parse/Lexer.cpp

```diff
@@ -1204,6 +1206,202 @@ static bool maybeConsumeNewlineEscape(const char *&CurPtr, ssize_t Offset) {
   }
 }
 
+const static uint32_t ZeroWidthC[] = {
```

Comment:
Should we extract a function to test for zero-width characters and put it in swift/Basic/Unicode.{h,cpp}?

I can't speak for the accuracy of this list of characters, but a lot of them seem to be visible but take up no extra space—for instance, I pulled out U+1A59 at random and it turned out to be a combining mark. Do we want to diagnose those kinds of characters?

Reply:
@xwu don’t think it shouldn't be included until we come up with a comprehensive solution for all of Swift so we’re probably going to have to back it out anyway. I say leave it in for now while we wait.